test/core.jl

```diff
@@ -4505,6 +4505,15 @@ for T in (Any, ValueWrapper)
     end
 end
 
+#test grow_end ccall directly since it's used in the C source
+for ET in [Nothing, Int, Union{Int, Nothing}, Any]
```

Comment:
Is this test related to the new intrinsic?

Reply:
yes (in that the refactoring in genericmemory.c initially broke this test)